src/main/java/se/sundsvall/seabloader/service/mapper/InvoiceMapper.java

```diff
@@ -0,0 +1,60 @@
+package se.sundsvall.seabloader.service.mapper;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.commons.lang3.exception.ExceptionUtils.getRootCauseMessage;
+import static se.sundsvall.seabloader.integration.db.model.enums.Status.FAILED;
+
+import java.io.StringReader;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.JAXBIntrospector;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParserFactory;
+import javax.xml.transform.sax.SAXSource;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import se.inexchange.generated.InExchangeInvoiceStatusType;
+import se.sundsvall.seabloader.integration.db.model.InvoiceEntity;
+
+public class InvoiceMapper {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(InvoiceMapper.class);
+
+	private InvoiceMapper() {}
+
+	public static InvoiceEntity toInvoiceEntity(final byte[] fileContent) {
+		final var xmlContent = new String(fileContent, UTF_8);
+
+		try {
+			final var inExchangeInvoice = toInExchangeInvoice(xmlContent);
+			return InvoiceEntity.create()
+				.withContent(xmlContent)
+				.withInvoiceId(String.valueOf(inExchangeInvoice.getInvoice().getInvoiceId()));
+
+		} catch (final Exception e) {
+			LOGGER.error("Error during deserialization of XML content", e);
+			return InvoiceEntity.create()
```

Comment:
https://www.mariadbtutorial.com/mariadb-index/mariadb-unique-index/

```
In the database world, null values are special because they are markers that indicate the missing information. Therefore, a null value doesn’t equal to any value, even itself.

MariaDB respects the distinctions of null values when it comes to the unique index. Simply put, you can have multiple null values in a column that has a unique index.
```